src/expr/node_manager_template.cpp

```diff
@@ -601,6 +620,22 @@ std::vector<TypeNode> NodeManager::mkMutualDatatypeTypesInternal(
     if (dtp->getNumParameters() == 0)
     {
       typeNode = mkTypeConst(DatatypeIndexConstant(index));
+      // if the datatype is a tuple, the type will be (TUPLE_TYPE ...)
+      if (dt.isTuple())
+      {
+        TypeNode dtt = typeNode;
+        const DTypeConstructor& dc = dt[0];
+        std::vector<TypeNode> tupleTypes;
+        for (size_t i = 0, nargs = dc.getNumArgs(); i < nargs; i++)
```

Comment:
Do we need this for loop instead of `TypeNode::.getTupleTypes()` which seems appropriate here?

Reply:
Thanks for pointing this out, actually, the converse: `getTupleTypes` should be simplified to just return the child types computed by this loop.  I'll simplify the PR.